pulp_ansible/app/tasks/deletion.py

```diff
@@ -0,0 +1,82 @@
+"""
+This module includes tasks related to deleting Content.
+
+You can remove Content from a Repository by making a new RepositoryVersion
+without the Content. If an API endpoint uses a Distribution which points to
+the latest_version of the Repository then the Content is unavailable,
+however it is not deleted.
+
+Content can only be deleted if it exists in no RepositoryVersion.
+
+Content cannot be removed from a RepositoryVersion since it is immutable.
+
+Pulp's orphan_cleanup task deletes any Content not part of a RepositoryVersion.
+"""
+
+import logging
+
+from pulp_ansible.app.models import Collection, CollectionVersion
+from pulpcore.plugin.tasking import add_and_remove, orphan_cleanup
+
+log = logging.getLogger(__name__)
+
+
+def _cleanup_old_versions(repo):
+    """Delete all the old versions of the given repository."""
+    for version in repo.versions.complete().order_by("-number")[1:]:
+        version.delete()
+
+
+def _remove_collection_version_from_repos(collection_version):
+    """Remove CollectionVersion from latest RepositoryVersion of each repo."""
+    for repo in collection_version.repositories.all():
+        add_and_remove(repo.pk, add_content_units=[], remove_content_units=[collection_version.pk])
+        _cleanup_old_versions(repo)
+
+
+def delete_collection_version(collection_version_pk):
+    """Task to delete CollectionVersion object.
+
+    Sequentially do the following in a single task:
+    1. Call _remove_collection_version_from_repos
+    2. Run orphan_cleanup to delete the CollectionVersion
+    3. Delete Collection if it has no more CollectionVersion
+    """
+    collection_version = CollectionVersion.objects.get(pk=collection_version_pk)
+    collection = collection_version.collection
+
+    _remove_collection_version_from_repos(collection_version)
+
+    log.info("Running orphan_cleanup to delete CollectionVersion object and artifact")
```

Comment:
All logging in pulp_ansible should be i8ln wrapped with gettext

Reply:
@newswangerd would you possibly we willing to fix the log statements?

Reply:
@bmbouter internationalization on logging is not recommended, logging must be consistent regardless of the language of the system. As logs are not for end user it would be better to keep in English only.

Reply:
I'm not familiar with that perspective, and I've always heard the reverse. If someone can't understand the logs how could they interpret them? How can we sort this out because if you look through Pulp's existing code it strives to i8ln all strings including log messages. Can you educate me some more on this?

Reply:
i18n is for front-ends, messages for the end user must be translated such as in API errors, CLI output, UI components.

Log audience are system administrators, auditing, developer when troubleshooting, if a support/qe/dev need to analyze a log taken from a customer running in French, the logs should be in English, consistent with the development language.

Reply:
@bmbouter imagine you take a bug issue from a customer running a system in Chinese, customer attaches log output. Isn't better to have log in English?

Reply:
I hear you on the perspective, but right now we've got two perspectives. What evidence and guidance can we look at to help guide us on this?

Reply:
you can even search codebase to find where log where printed

Reply:
Ultimately who are logs really for though? I really think rather than debating one over the other we should try to find out what other large, multi-lingual apps do. What does the python community expect. How about GNU?

Reply:
Can you please bring this to discourse.pulpproject.org/ or to [open floor](https://hackmd.io/SVCMjpwXTfOMqF2OeyyLRw) because if what you're saying is right we should be doing this differently in all of Pulp. That's a good maturity point for an application, but we shouldn't really debate it here. Also, from a consistency perspective this is inconsistent to how it's been done broadly to date.

Reply:
@bmbouter the utility of logs are for debugging, auditing, monitoring, usually those professionals prefer English so it is consistent among different applications.

Log levels for example are "ERROR" or "WARNING" standardized and not "ERRO" "AVISO" if I switch my app front end to Portuguese.

Reply:
even log viewers tools filters by messages like ERROR, CRITICAL and there are tools that can do other kinds of filtering and analysis using English as base.

Reply:
Maybe I haven't been clear, I see your point. Convincing me does very little though. Can you help me bring this to the larger audience of 20+ Pulp devs? Also can we stay consistent with the current style until we have a broad understanding that it's changed?

Reply:
+1 to bring to a broader audience.